static/src/javascripts/lib/cmp.spec.js

```diff
@@ -0,0 +1,215 @@
+// @flow
+import { init, onConsentNotification, consentState, _ } from 'lib/cmp';
+import { getAdConsentState as _getAdConsentState } from 'common/modules/commercial/ad-prefs.lib';
+
+const getAdConsentState: any = _getAdConsentState;
+
+jest.mock('common/modules/commercial/ad-prefs.lib', () => ({
+    getAdConsentState: jest.fn(),
+    thirdPartyTrackingAdConsent: jest.fn(),
+}));
+
+describe('cmp', () => {
+    beforeEach(() => {
+        _.resetCmp();
+        getAdConsentState.mockReset();
+    });
+
+    it('does not re-trigger consent notifications if init called multiple times', () => {
+        const myCallBack = jest.fn();
+
+        init();
+
+        onConsentNotification('functional', myCallBack);
+
+        expect(myCallBack).toHaveBeenCalledTimes(1);
+
+        init();
+
+        expect(myCallBack).toHaveBeenCalledTimes(1);
+    });
+
+    describe('consentState', () => {
+        it('returns null state if cmp lib has not been initialised', () => {
+            expect(consentState('functional')).toBeNull();
+            expect(consentState('performance')).toBeNull();
+            expect(consentState('advertisement')).toBeNull();
+        });
+
+        it('returns functional consent state', () => {
+            init();
+
+            expect(consentState('functional')).toBe(true);
```

Comment:
I still don't like these being hard-coded to true here, but I do see why that's the case. These tests will need to depend on the current state of the user's consent. Of course, right now we don't ask about functional so the current consent state is irrelevant, it just means these tests will be changing in the future. It's ok to leave this as is as long as we're careful to make sure this gets fixed later!

Reply:
This'll certainly be fixed in the code which will in turn cause this test to become out of date at which point we'll definitely need to fix the test.